docs/mvc/tutorials/mvc-with-entity-framework/sample/src/ContosoBooks/Startup.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Diagnostics;
+using Microsoft.AspNet.Hosting;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.ConfigurationModel;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Console;
+using ContosoBooks.Models;
+using Microsoft.Data.Entity;
+
+namespace ContosoBooks
+{
+    public class Startup
+    {
+        public Startup(IHostingEnvironment env)
+        {
+            // Setup configuration sources.
+            Configuration = new Configuration()
+                .AddJsonFile("config.json")
+                .AddEnvironmentVariables();
+        }
+
+        public IConfiguration Configuration { get; set; }
+
+        // This method gets called by the runtime.
+        public void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<AppSettings>(Configuration.GetSubKey("AppSettings"));
+
+            // Add MVC services to the services container.
+            services.AddMvc();
+
+            services.AddEntityFramework()
+                //.AddInMemoryStore()
+                .AddSqlServer()
+                .AddDbContext<BookContext>(options =>
+                {
+                    options.UseSqlServer(Configuration.Get("Data:ConnectionString"));
+                });
+
```

Comment:
If you wanted to you could just leave AddInMemory() here and the you can just swap the UseSqlServer for UseInMemory to swap.

```csharp
            services.AddEntityFramework()
                .AddInMemoryStore()
                .AddSqlServer()
                .AddDbContext<BookContext>(options =>
                {
                    // options.UseInMemory();
                    options.UseSqlServer(Configuration.Get("Data:ConnectionString"));
                });
```